impl/src/planning/subplanners.rs

```diff
@@ -264,59 +280,40 @@ impl<'planner> CrateSubplanner<'planner> {
     let mut targets = self.produce_targets(&package_root)?;
     let build_script_target_opt = self.take_build_script_target(&mut targets);
 
-    let mut lib_target_name = None;
-    let mut is_proc_macro = false;
-    {
-      for target in &targets {
-        if target.kind == "lib" {
-          lib_target_name = Some(target.name.clone());
-          break;
-        }
-        if target.kind == "proc-macro" {
-          is_proc_macro = true;
-          lib_target_name = Some(target.name.clone());
-          break;
-        }
-      }
-    }
-
-    // Build a list of dependencies while addression a potential whitelist of target triples
-    let mut filtered_deps = Vec::new();
-    for dep_set in targeted_deps.iter() {
-      let mut target_triples = util::get_matching_bazel_triples(&dep_set.target)?;
-      util::filter_bazel_triples(
-        &mut target_triples,
-        self
-          .settings
-          .targets
-          .as_ref()
-          .unwrap_or(&Vec::<String>::new()),
-      );
-
-      if target_triples.is_empty() {
-        continue;
-      }
-
-      filtered_deps.push(CrateTargetedDepContext {
-        target: dep_set.target.clone(),
-        deps: CrateDependencyContext {
-          dependencies: dep_set.dependencies.normal_deps.clone(),
-          proc_macro_dependencies: dep_set.dependencies.proc_macro_deps.clone(),
-          data_dependencies: vec![],
-          build_dependencies: dep_set.dependencies.build_deps.clone(),
-          build_proc_macro_dependencies: dep_set.dependencies.build_proc_macro_deps.clone(),
-          build_data_dependencies: vec![],
-          dev_dependencies: dep_set.dependencies.dev_deps.clone(),
-          aliased_dependencies: dep_set.dependencies.aliased_deps.clone(),
-        },
-        conditions: util::generate_bazel_conditions(
-          &self.settings.rust_rules_workspace_name,
-          &target_triples,
-        )?,
-      });
-    }
+    let lib_target_name = targets
+      .iter()
+      .find(|target| target.kind == "lib" || target.kind == "proc-macro")
+      .map(|target| target.name.clone());
+
+    let is_proc_macro = targets.iter().any(|target| target.kind == "proc_macro");
```

Comment:
```suggestion
    let is_proc_macro = targets.iter().any(|target| target.kind == "proc-macro");
```

Aren't string literals fun? 😄